pygmt/src/grdfill.py

```diff
@@ -0,0 +1,77 @@
+"""
+grdfill - Fill blank areas from a grid.
+"""
+
+import xarray as xr
+from pygmt.clib import Session
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+
+
+@fmt_docstring
+@use_alias(
+    A="mode",
+    G="outgrid",
+    R="region",
+)
+@kwargs_to_strings(R="sequence")
+def grdfill(grid, **kwargs):
+    r"""
+    Fill blank areas from a grid file.
+
+    Read a grid that presumably has unfilled holes that the user
+    wants to fill in some fashion.  Holes are identified by NaN values but
+    this criteria can be changed.  There are several different algorithms that
+    can be used to replace the hole values.
+
+    Full option list at :gmt-docs:`grdfill.html`
+
+    {aliases}
+
+    Parameters
+    ----------
+    grid : str or xarray.DataArray
+        The file name of the input grid or the grid loaded as a DataArray.
+    outgrid : str or None
+        The name of the output netCDF file with extension .nc to store the grid
+        in.
+    mode : str
+        Specify the hole-filling algorithm to use.  Choose from **c** for
+        constant fill and append the constant value, **n** for nearest
+        neighbor (and optionally append a search radius in
+        pixels [default radius is `r^2 = sqrt(X^2 + Y^2)`,
```

Comment:
LaTeX formula?
```suggestion
        pixels [default radius is :math:`r^2 = sqrt(X^2 + Y^2)`,
```

Reply:
If you want to use the `math` directive, it should be:
```
:math:`r^2 = \sqrt{X^2 + Y^2}`
```

Reply:
> If you want to use the `math` directive, it should be:
> 
> ```
> :math:`r^2 = \sqrt{X^2 + Y^2}`
> ```

updated~ Thanks

Reply:
@willschlitzer previously had this in there - it did not work because `fmt_docstring` tries to insert common text into the brackets. So, it either needs to stay as non-sphinx math or `fmt_docstring` needs to be modified.

Reply:
If we want LaTeX math, we may need to use double curly braces `:math:`r^2 = \sqrt{{ X^2 + Y^2 }}`` . See https://stackoverflow.com/questions/5466451

Reply:
> If we want LaTeX math, we may need to use double curly braces
> 
> ```
> :math:`r^2 = \sqrt{{ X^2 + Y^2 }}`
> ```
> 
> See https://stackoverflow.com/questions/5466451

This appears to be working!